ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/feemarket/CancunFeeMarket.java

```diff
@@ -26,11 +27,24 @@
 public class CancunFeeMarket extends LondonFeeMarket {
   private static final Logger LOG = LoggerFactory.getLogger(CancunFeeMarket.class);
   protected static final BigInteger BLOB_GAS_PRICE = BigInteger.ONE;
-  private static final BigInteger BLOB_GAS_PRICE_UPDATE_FRACTION = BigInteger.valueOf(3338477);
+
+  protected final BigInteger baseFeeUpdateFraction;
 
   public CancunFeeMarket(
-      final long londonForkBlockNumber, final Optional<Wei> baseFeePerGasOverride) {
+      final long londonForkBlockNumber,
+      final Optional<Wei> baseFeePerGasOverride,
+      final long baseFeeUpdateFraction) {
     super(londonForkBlockNumber, baseFeePerGasOverride);
+
+    this.baseFeeUpdateFraction = BigInteger.valueOf(baseFeeUpdateFraction);
+  }
+
+  public CancunFeeMarket(
```

Comment:
out of scope for this PR really, but I think this should be package-private and the callers should call FeeMarket.cancun

Reply:
Sounds like a good idea. I made all the FeeMarkets package private and updated all the places that used to constructors to use the factories instead